### ISSUE TITLE
Audit coobook via Chef Automate fails to inherit profiles (#<TypeError: no implicit conversion of URI::HTTPS into String>)

### ISSUE THREAD

**Issue by timusmc96** [chef-cookbooks/audit#222]
### Cookbook version
3.1

### Chef-client version
12.18.31

### Platform Details
windows 2012R2

### Scenario:
run audit against window server with an inherit profile and get this error

#<TypeError: no implicit conversion of URI::HTTPS into String>

### Steps to Reproduce:
[If you are filing an issue what are the things we need to do in order to repro your problem? How are you using this cookbook or any resources it includes?]

### Expected Result:
Compliance Should complete

### Actual Result:
```
***********************************************
C:\opscode\chef\embedded\bin>chef-client
Starting Chef Client, version 12.18.31
resolving cookbooks for run list: ["audit"]
Synchronizing Cookbooks:
  - audit (3.1.0)
  - compat_resource (12.16.3)
Installing Cookbook Gems:
Compiling Cookbooks...
Recipe: audit::inspec
  * inspec_gem[inspec] action install (up to date)
  Converging 1 resources
  * inspec_gem[inspec] action nothing (skipped due to action :nothing)

Running handlers:
[2017-05-10T17:26:31-04:00] WARN: Format is json
[2017-05-10T17:26:32-04:00] ERROR: Report handler Chef::Handler::AuditReport rai
sed #<TypeError: no implicit conversion of URI::HTTPS into String>
[2017-05-10T17:26:32-04:00] ERROR: C:/chef/cache/cookbooks/audit/files/default/v
endor/chef-server/fetcher.rb:115:in `match'
[2017-05-10T17:26:32-04:00] ERROR: C:/chef/cache/cookbooks/audit/files/default/v
endor/chef-server/fetcher.rb:115:in `chef_server_url'
[2017-05-10T17:26:32-04:00] ERROR: C:/chef/cache/cookbooks/audit/files/default/v
endor/chef-server/fetcher.rb:74:in `resolved_source'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/inspec-1.23.0/lib/inspec/cached_fetcher.rb:52:in `assert_cache_sanity!'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/inspec-1.23.0/lib/inspec/cached_fetcher.rb:45:in `fetch'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/inspec-1.23.0/lib/inspec/profile.rb:69:in `for_fetcher'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/inspec-1.23.0/lib/inspec/dependencies/requirement.rb:107:in `profile'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/inspec-1.23.0/lib/inspec/dependencies/dependency_set.rb:65:in `block in eac
h'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/inspec-1.23.0/lib/inspec/dependencies/dependency_set.rb:64:in `each'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/inspec-1.23.0/lib/inspec/dependencies/dependency_set.rb:64:in `each'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/inspec-1.23.0/lib/inspec/profile.rb:167:in `load_libraries'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/inspec-1.23.0/lib/inspec/runner.rb:83:in `block in load'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/inspec-1.23.0/lib/inspec/runner.rb:79:in `each'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/inspec-1.23.0/lib/inspec/runner.rb:79:in `load'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/inspec-1.23.0/lib/inspec/runner.rb:100:in `run'
[2017-05-10T17:26:32-04:00] ERROR: C:/chef/cache/cookbooks/audit/files/default/h
andler/audit_report.rb:128:in `call'
[2017-05-10T17:26:32-04:00] ERROR: C:/chef/cache/cookbooks/audit/files/default/h
andler/audit_report.rb:54:in `report'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/chef-12.18.31-universal-mingw32/lib/chef/handler.rb:259:in `run_report_unsa
fe'
[2017-05-10T17:26:32-04:00] ERROR: C:/chef/cache/cookbooks/audit/files/default/h
andler/audit_report.rb:72:in `run_report_safely'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/chef-12.18.31-universal-mingw32/lib/chef/handler.rb:125:in `block in run_re
port_handlers'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/chef-12.18.31-universal-mingw32/lib/chef/handler.rb:123:in `each'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/chef-12.18.31-universal-mingw32/lib/chef/handler.rb:123:in `run_report_hand
lers'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/chef-12.18.31-universal-mingw32/lib/chef/handler.rb:135:in `block in <class
:Handler>'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/chef-12.18.31-universal-mingw32/lib/chef/client.rb:441:in `block in run_com
pleted_successfully'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/chef-12.18.31-universal-mingw32/lib/chef/client.rb:440:in `each'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/chef-12.18.31-universal-mingw32/lib/chef/client.rb:440:in `run_completed_su
ccessfully'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/chef-12.18.31-universal-mingw32/lib/chef/client.rb:299:in `run'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/chef-12.18.31-universal-mingw32/lib/chef/application.rb:277:in `run_with_gr
aceful_exit_option'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/chef-12.18.31-universal-mingw32/lib/chef/application.rb:253:in `block in ru
n_chef_client'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/chef-12.18.31-universal-mingw32/lib/chef/local_mode.rb:44:in `with_server_c
onnectivity'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/chef-12.18.31-universal-mingw32/lib/chef/application.rb:236:in `run_chef_cl
ient'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/chef-12.18.31-universal-mingw32/lib/chef/application/client.rb:464:in `slee
p_then_run_chef_client'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/chef-12.18.31-universal-mingw32/lib/chef/application/client.rb:451:in `bloc
k in interval_run_chef_client'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/chef-12.18.31-universal-mingw32/lib/chef/application/client.rb:450:in `loop
'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/chef-12.18.31-universal-mingw32/lib/chef/application/client.rb:450:in `inte
rval_run_chef_client'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/chef-12.18.31-universal-mingw32/lib/chef/application/client.rb:434:in `run_
application'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/chef-12.18.31-universal-mingw32/lib/chef/application.rb:59:in `run'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/lib/ruby/gems/2.3.0/
gems/chef-12.18.31-universal-mingw32/bin/chef-client:26:in `<top (required)>'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/bin/chef-client:22:i
n `load'
[2017-05-10T17:26:32-04:00] ERROR: C:/opscode/chef/embedded/bin/chef-client:22:i
n `<main>'
  - Chef::Handler::AuditReport
Running handlers complete
Chef Client finished, 0/2 resources updated in 12 seconds
```
***************************

**Comment by timusmc96** [comment]
Chris looks like this issue is not fixed with windows.   Basically today I was able to get all profiles uploaded into Chef Automate with the one user ID, all run fine, except the windows one that inherits profile.  If we can get on a call that would be great.

**Comment by chris-rock** [comment]
Seems to be a different error that the tar issue. I have never seen that before.

**Comment by chris-rock** [comment]
The question is how @target in https://github.com/chef-cookbooks/audit/blob/master/files/default/vendor/chef-server/fetcher.rb#L115 can be an URI

**Comment by timusmc96** [comment]
Let me know if you have time to look into this Monday am.  thanks

**Comment by timusmc96** [comment]
Any chance to look into this issue?

**Comment by chris-rock** [comment]
Not right now ;-)

**Comment by kekaichinose** [comment]
Thank you for tracking down this issue. As with many large open source projects, we have a backlog of bugs and issues of varying user impact. Some issues are perpetually subject to lower priority than others in our backlog. We have reviewed the issue, and we have determined this a very low priority when considered in the context of the entire backlog. We have elected to close it as "Declined" due to competing priorities. We very much appreciate you bringing this to our attention.

Thank you!
__Chef InSpec Team__
